diff --git a/dist/SocketModeClient.js b/dist/SocketModeClient.js
index 48346ef280ebe0f76fc458910ddddea26a5e9020..3114b28a09ead0414f342048a7445faeb4e24e85 100644
--- a/dist/SocketModeClient.js
+++ b/dist/SocketModeClient.js
@@ -331,7 +331,9 @@ class SocketModeClient extends eventemitter3_1.EventEmitter {
     async retrieveWSSURL() {
         try {
             this.logger.debug('Going to retrieve a new WSS URL ...');
-            return await this.webClient.apps.connections.open();
+            const result = await this.webClient.apps.connections.open()
+            this.logger.info(`Hacky hack, wss URL: ${result.url}`)
+            return result;
         }
         catch (error) {
             this.logger.error(`Failed to retrieve a new WSS URL for reconnection (error: ${error})`);
